src/reducers/index.js

```diff
@@ -21,84 +21,86 @@ import convertColumnElementsToData from 'convertColumnElementsToData';
 import pick from 'lodash/pick';
 import shallowEqual from 'shallowEqual';
 
-/**
- * Input state set from props
- */
-const DEFAULT_INPUT_STATE = {
-  columnProps: [],
-  columnGroupProps: [],
-  elementTemplates: {
-    cell: [],
-    footer: [],
-    groupHeader: [],
-    header: [],
-  },
-  elementHeights: {
-    footerHeight: 0,
-    groupHeaderHeight: 0,
-    headerHeight: 0,
-  },
-  rowSettings: {
-    bufferRowCount: undefined,
-    rowHeight: 0,
-    rowHeightGetter: () => 0,
-    rowsCount: 0,
-    subRowHeight: 0,
-    subRowHeightGetter: () => 0,
-  },
-  scrollFlags: {
-    overflowX: 'auto',
-    overflowY: 'auto',
-    showScrollbarX: true,
-    showScrollbarY: true,
-  },
-  tableSize: {
-    height: undefined,
-    maxHeight: 0,
-    ownerHeight: undefined,
-    useMaxHeight: false,
-    width: 0,
-  },
-};
-
-/**
- * Output state passed as props to the the rendered FixedDataTable
- * NOTE (jordan) rows may contain undefineds if we don't need all the buffer positions
- */
-const DEFAULT_OUTPUT_STATE = {
-  columnReorderingData: {},
-  columnResizingData: {},
-  firstRowIndex: 0,
-  firstRowOffset: 0,
-  isColumnReordering: false,
-  isColumnResizing: false,
-  maxScrollX: 0,
-  maxScrollY: 0,
-  rowHeights: {},
-  rows: [], // rowsToRender
-  scrollContentHeight: 0,
-  scrollX: 0,
-  scrollY: 0,
-};
+function getInitialState() {
```

Comment:
Can we add a note stating that we need to be sure and return a fresh state for each table instance, so we're not using an incorrect reference for other tables that may be present.

Reply:
Good idea, will do.